### PR TITLE
[CORE-1459] Add tooling to copy function/edge function scripts to dist folder

### PR DESCRIPTION
**Pull request by RoboCafaz**
Added a very simple bit of functionality to the netlify-webpack-plugin that will allow you to specify files to be used as functions/edge functions. Doesn't do any transpilation or anything, just copies the files exactly as-is.

Also switched to GH actions so we can actually deploy this when it's ready.

### CODE REVIEW

File: .github/workflows/build-and-test.yaml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+      - name: Use Node.js
+        uses: actions/setup-node@v2
+        with:
+          node-version-file: ".nvmrc"
+          cache: "yarn"
+
+      - name: Yarn Install
+        run: yarn install
+        shell: bash
```
Comment (kisaraofpern) [inline .github/workflows/build-and-test.yaml:27]: Ah! This option would've helped me in my stack-lib PR.

File: scripts/deploy.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
+const fs = require('fs');
```
Comment (kisaraofpern) [inline scripts/deploy.js:1]: Is there a way to test this/validate this locally?

File: example/src/edge-functions/example.ts [hunk 3]
```diff
@@ -0,0 +1,3 @@
+export default () => new Response('Goodnight, moon!');
```
Comment (kisaraofpern) [inline example/src/edge-functions/example.ts:1]: Fun fact:
https://www.fatherly.com/news/goodnight-moon-new-york-public-library-history

File: jest.config.js [hunk 4]
```diff
@@ -5,7 +5,7 @@ module.exports = {
   coveragePathIgnorePatterns: ['__(.+?)__'],
   verbose: true,
   rootDir: join(__dirname, './src'),
-  moduleFileExtensions: ['js','ts'],
+  moduleFileExtensions: ['js', 'ts'],
```
Comment (kisaraofpern) [inline jest.config.js:8]: Nit: Separate out prettier changes.

File: scripts/deploy.js (hunk 2 above)
Comment (RoboCafaz) [reply scripts/deploy.js:1]: It's pretty rare to see us write tests for our scripts. But I have verified this is functioning by running it locally and then unpublishing the version of package.

**Review by kisaraofpern — APPROVED** [review]

File: jest.config.js (hunk 4 above)
Comment (RoboCafaz) [reply jest.config.js:8]: Done.

**Review by kstrickland0612 — APPROVED** [review]

**Pull request merged**